### ISSUE TITLE
Named backend server instead of generic SRV_x name

### ISSUE THREAD

**Issue by pawellrus** [haproxytech/kubernetes-ingress#662]
Hello.
I'm interesting, is it possible to define custom backend downstream server names instead of generic naming SRV_x. For example it could be the pod name or another custom value passed by annotation.

This is critical in some kind of scenarios like rolling update with putting backend server for maint. Now in UI it's not possible to understand which backend will go to maint.

**Comment by fabianonunes** [comment]
I don't think this is possible today: https://github.com/haproxy/haproxy/issues/952

**Comment by pawellrus** [comment]
@fabianonunes i'm not talking about renaming in runtime. i'm talking about correct naming before populatin backend configuration.
Btw may be there is another way to diffrentiate backend servers during runtime. Via labels or something like that?

**Comment by pawellrus** [comment]
@ivanmatmati so it's impossible to put selected pod to maintenance before dis-labeling them to make sure that users' sessions won't broke forcefully?

**Comment by hoerup** [comment]
wouldn't it be possible to use "add server backend-name/podname ip:port" for new entries and "del server backend-name/podname" when removing ?

**Comment by ivanmatmati** [comment]
@pawellrus , the servers list the ip address of the pod so you can always find the original pod.

**Comment by oktalz** [comment]
@mike-code the ones that are marked for maint are the ones that are marked from kubernetes to be soon removed.

@hoerup 
about
> wouldn't it be possible to use "add server backend-name/podname ip:port" for new entries and "del server backend-name/podname" when removing ?

this is in roadmap, its a bit more complicated that would would think initially, but we do have that in plans
